File: HackerRank/30DaysOfCode/Day10_Binary_Numbers.cs
using System;
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;


namespace HackerRank
{
    class Day10_Binary_Numbers
    {
        public static void BinaryNumbers()
        {

            // binary  0| 1 | 2 | 4 | 8 |16 |32 |64 |128
            // binary2 0|  0| 1 | 2 |  3| 4 |  5| 6 | 7

            //Binary:    r:     0       1      0     0       0       1   flip: 100010 =34
            // Base 10: 34 | 2/34=17 2/17=8  2/8=4  2/4=2   2/2=1   2/1=0
            //int[] binaryNum = { 1, 0, 1 ,1}; // Bsse 10 = 13 ,  2 consecitive 1s.

            List<int> binaryNum = new List<int>();
            Console.WriteLine("Input a number to be converted to binary");
            int base10 = Convert.ToInt32(Console.ReadLine());
            int counter=0;
            int total = 0;
            //int[] count = new int[0];
            List<int> count = new List<int>();

            while (base10>0)
            {
                int quotient = base10 /2;
                int remainder = base10 % 2;
                base10 = quotient;
               //   Console.WriteLine("remain: {0}",remainder);
                binaryNum.Add(remainder);
                
            } 
           binaryNum.Reverse();
            


            for (int index = 0; index < binaryNum.Count; index++)
            {

                if (binaryNum[index] == 1)
                {
                    counter++;
                    total = counter;
                    count.Add(total);
                    //count[index] = total;
                }
                else
                    counter  = 0;

                //if(total > index)

             //   Console.WriteLine("Index{0}:   {1}", index, binaryNum[index]);
                
            }
            count.Sort();
            count.Reverse();


            foreach (int item in count)
                Console.Write("{0}, ", item);
            Console.WriteLine();
            foreach (int number in binaryNum)
                Console.Write(number);

            Console.WriteLine("\n--------------");
            Console.WriteLine("\n\nCount:  {0}", count[0]);
        }
    }
}

File: java/30DaysOfCoding/Day10_Binary_Numbers.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

         static void binaryNumbers(int baseTen){

            List<Integer> binaryNum = new ArrayList<>();
            int counter=0;
             int total = 0;
             List<Integer> count = new ArrayList<>();

             while (baseTen>0){
                int quotient = baseTen /2;
                int remainder = baseTen % 2;
                baseTen = quotient;
                binaryNum.add(remainder);
            } 
            Collections.reverse(binaryNum);

            for (int index = 0; index < binaryNum.size(); index++)
            {
                if (binaryNum.get(index) == 1)
                {
                    counter++;
                    total = counter;
                    count.add(total);
                }
                else
                    counter  = 0;
            }
            Collections.sort(count);
            Collections.reverse(count);
            System.out.print(count.get(0));
        }
    

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
         binaryNumbers(n);
        scanner.close();
    }
}
